src/commands/createFunction/JobsListStep.ts

```diff
@@ -3,49 +3,71 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizardExecuteStep, AzureWizardPromptStep, IWizardOptions } from '@microsoft/vscode-azext-utils';
+import { AzureWizardExecuteStep, AzureWizardPromptStep, IAzureQuickPickItem, IWizardOptions } from '@microsoft/vscode-azext-utils';
 import { localize } from '../../localize';
+import { JobType, ParsedJob } from '../../templates/script/parseScriptTemplatesV2';
+import { ScriptProjectCreateStep } from '../createNewProject/ProjectCreateStep/ScriptProjectCreateStep';
+import { isFunctionProject } from '../createNewProject/verifyIsProject';
+import { FunctionWizardV2Context } from './FunctionV2WizardContext';
 import { IFunctionWizardContext } from './IFunctionWizardContext';
+import { actionStepFactory } from './actionStepsV2/actionStepFactory';
+import { promptStepFactory } from './promptStepsV2/promptStepFactory';
 
 /* eslint-disable @typescript-eslint/no-non-null-assertion */
 export class JobsListStep extends AzureWizardPromptStep<IFunctionWizardContext> {
     public constructor(readonly isProjectWizard?: boolean) {
         super();
     }
 
-    public async prompt(context: IFunctionWizardContext): Promise<void> {
+    public async prompt(context: FunctionWizardV2Context): Promise<void> {
         // TODO: come up with a better placeholder
         const placeHolder: string = localize('selectJob', 'Select a job');
-        context.job = (await context.ui.showQuickPick(context.functionTemplateV2!.wizards.map((w) => { return { label: w.name, data: w } }), { placeHolder })).data;
+        context.job = (await context.ui.showQuickPick(this.getPicks(context), { placeHolder })).data;
     }
 
-    public shouldPrompt(context: IFunctionWizardContext): boolean {
+    public shouldPrompt(context: FunctionWizardV2Context): boolean {
         return !context.job && !!context.functionTemplateV2;
     }
 
-    public async getSubWizard(context: IFunctionWizardContext): Promise<IWizardOptions<IFunctionWizardContext> | undefined> {
-        // create steps based on the job
+    public async getSubWizard(context: FunctionWizardV2Context): Promise<IWizardOptions<FunctionWizardV2Context> | undefined> {
         if (context.job) {
-            const promptSteps: AzureWizardPromptStep<IFunctionWizardContext>[] = [];
-            context.job.prompts.map(_p => {
-                promptSteps.push();
+            const promptSteps: AzureWizardPromptStep<FunctionWizardV2Context>[] = [];
+            context.job.prompts.map(p => {
+                promptSteps.push(promptStepFactory(p));
             });
```

Comment:
This code will also make more sense IMO if `prompts` is renamed to `parsedInputs`:

```suggestion
            context.job.parsedInputs.map(parsedInput => {
                promptSteps.push(promptStepFactory(parsedInput));
            });
```